pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ConsumerHandler.java

```diff
@@ -69,35 +69,35 @@
 
     private final int maxPendingMessages;
     private final AtomicInteger pendingMessages = new AtomicInteger();
-    
+
     private final LongAdder numMsgsDelivered;
     private final LongAdder numBytesDelivered;
     private final LongAdder numMsgsAcked;
     private volatile long msgDeliveredCounter = 0;
     private static final AtomicLongFieldUpdater<ConsumerHandler> MSG_DELIVERED_COUNTER_UPDATER =
             AtomicLongFieldUpdater.newUpdater(ConsumerHandler.class, "msgDeliveredCounter");
 
-    public ConsumerHandler(WebSocketService service, HttpServletRequest request) {
-        super(service, request);
+    public ConsumerHandler(WebSocketService service, HttpServletRequest request, ServletUpgradeResponse response) {
+        super(service, request, response);
         this.subscription = extractSubscription(request);
         this.conf = getConsumerConfiguration();
         this.maxPendingMessages = (conf.getReceiverQueueSize() == 0) ? 1 : conf.getReceiverQueueSize();
         this.numMsgsDelivered = new LongAdder();
         this.numBytesDelivered = new LongAdder();
         this.numMsgsAcked = new LongAdder();
-        
-    }
 
-    @Override
-    protected void createClient(Session session) {
         try {
             this.consumer = service.getPulsarClient().subscribe(topic, subscription, conf);
             this.service.addConsumer(this);
             receiveMessage();
         } catch (Exception e) {
-            log.warn("[{}] Failed in creating subscription {} on topic {}", session.getRemoteAddress(), subscription,
+            log.warn("[{}] Failed in creating subscription {} on topic {}", request.getRemoteAddr(), subscription,
                     topic, e);
-            close(WebSocketError.FailedToSubscribe, e.getMessage());
+            try {
+                response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Failed to subscribe");
+            } catch (IOException e1) {
+                log.warn("Failed to send error: {}", e1);
```

Comment:
`log.warn("Failed to send error: {}", e1.getMessage());`
or can we add more info to help for which subscription it has received error.
`log.warn("[{}] Failed to send error: {}", request.getRemoteAddr(), subscribe,  e1);`

Reply:
Fixed